tests/lib/rules/no-restricted-imports.js

```diff
@@ -1020,6 +1020,199 @@ ruleTester.run("no-restricted-imports", rule, {
+        errors: [
+            {
+                message: "'Image' import from 'react-native' is restricted. import Image from ui/_components instead",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 10,
+                endColumn: 15
+            },
+            {
+                message: "'Text' import from 'react-native' is restricted. import Text from ui/_components instead",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 17,
+                endColumn: 21
+            },
+            {
+                message: "'ScrollView' import from 'react-native' is restricted. import ScrollView from ui/_components instead",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 23,
+                endColumn: 33
+            }
+        ]
+    },
+    {
+        code: "import { foo, bar, baz } from 'mod'",
+        options: [{
+            paths: [
+                {
+                    name: "mod",
+                    importNames: ["foo"],
+                    message: "Import foo from qux instead."
+                },
+                {
+                    name: "mod",
+                    importNames: ["baz"],
+                    message: "Import baz from qux instead."
+                }
+            ]
+        }],
+        errors: [
+            {
+                message: "'foo' import from 'mod' is restricted. Import foo from qux instead.",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 10,
+                endColumn: 13
+            },
+            {
+                message: "'baz' import from 'mod' is restricted. Import baz from qux instead.",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 20,
+                endColumn: 23
+            }
+        ]
+    },
+    {
+        code: "import * as mod from 'mod'",
+        options: [{
+            paths: [
+                {
+                    name: "mod",
+                    importNames: ["foo"],
+                    message: "Import foo from qux instead."
+                },
+                {
+                    name: "mod",
+                    importNames: ["bar"],
+                    message: "Import bar from qux instead."
+                }
+            ]
+        }],
+        errors: [
+            {
+                message: "* import is invalid because 'foo' from 'mod' is restricted. Import foo from qux instead.",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 8,
+                endColumn: 16
+            },
+            {
+                message: "* import is invalid because 'bar' from 'mod' is restricted. Import bar from qux instead.",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 8,
+                endColumn: 16
+            }
+        ]
+    },
```

Comment:
Perhaps this could be a single error with a concatenated message, but that would be complicated as these texts are generated by interpolating `data`, and I'm not sure if one big message would be better than multiple errors.

Reply:
I think it's fine the way it is. If people are configuring the rule this way then we can let them decide to change how it's configured if this isn't the behavior they want.